src/test/ui/unused/unused-field-assign.stderr

```diff
@@ -0,0 +1,119 @@
+error: unused assignement to temporary
+  --> $DIR/unused-field-assign.rs:11:20
+   |
+LL |     Foo { x: 2 }.x = 3;
+   |     ------------   ^
+   |     |
+   |     this is not bound to any variable
+   |
+note: the lint level is defined here
+  --> $DIR/unused-field-assign.rs:1:9
+   |
+LL | #![deny(unused_assignments)]
+   |         ^^^^^^^^^^^^^^^^^^
+help: consider assigning this struct to a local variable
+  --> $DIR/unused-field-assign.rs:11:5
+   |
+LL |     Foo { x: 2 }.x = 3;
+   |     ^^^^^^^^^^^^
+
+error: unused assignement to temporary
+  --> $DIR/unused-field-assign.rs:14:20
+   |
+LL |     Foo { x: 2 }.x += 1;
+   |     ------------   ^^
+   |     |
+   |     this is not bound to any variable
+   |
+help: consider assigning this struct to a local variable
+  --> $DIR/unused-field-assign.rs:14:5
+   |
+LL |     Foo { x: 2 }.x += 1;
+   |     ^^^^^^^^^^^^
+
+error: unused assignement to temporary
+  --> $DIR/unused-field-assign.rs:18:15
+   |
+LL |     (2, 12).0 += 10;
+   |     -------   ^^
+   |     |
+   |     this is not bound to any variable
+   |
+help: consider assigning this tuple to a local variable
+  --> $DIR/unused-field-assign.rs:18:5
+   |
+LL |     (2, 12).0 += 10;
+   |     ^^^^^^^
+
+error: unused assignement to temporary
+  --> $DIR/unused-field-assign.rs:21:19
+   |
+LL |     (10, false).1 = true;
+   |     -----------   ^
+   |     |
+   |     this is not bound to any variable
+   |
+help: consider assigning this tuple to a local variable
+  --> $DIR/unused-field-assign.rs:21:5
+   |
+LL |     (10, false).1 = true;
+   |     ^^^^^^^^^^^
+
+error: unused assignement to temporary
+  --> $DIR/unused-field-assign.rs:25:11
+   |
+LL |     FOO.x = 2;
+   |     ---   ^
+   |     |
+   |     this is not bound to any variable
+   |
+help: consider assigning this to a local variable
+  --> $DIR/unused-field-assign.rs:25:5
+   |
+LL |     FOO.x = 2;
+   |     ^^^
```

Comment:
We could make this a `structured_suggestion` for `let mut foo = FOO; foo.x = 2;`.